### PR TITLE
Use addImmediateErrorNotification

### LINKED ISSUE #1445: ImmediateErrorNotification are not shown
More specifically, the `mapVault.installationFailed` message is not shown when map installation fails.

Code that handles exceptions should always **log the exception as well** due to possibility of bugs in notification code, not just use notifications. Somebody should search the code for all uses of `ImmediateErrorNotification ` and add missing logging statements.

OS: Linux

### PR DESCRIPTION
**Pull request by Sheikah45**
Fixes #1445 

Use addImmediateErrorNotification for all Error Notifications and add logging before.

### CODE REVIEW

**Comment by Sheikah45** [comment]
anything to do here?

**Review by 1-alex98 — DISMISSED** [review]

**Comment by 1-alex98** [comment]
Yes rebase @Sheikah45

File: src/main/java/com/faforever/client/patch/GameUpdaterImpl.java [hunk 1]
```diff
@@ -82,6 +82,7 @@ public GameUpdater addFeaturedModUpdater(FeaturedModUpdater featuredModUpdater)
     return future
         .thenCompose(s -> updateGameBinaries(patchResults.get(patchResults.size() - 1).getVersion()))
         .exceptionally(throwable -> {
+          log.warn("Game not terminated correctly", throwable);
```
Comment (Brutus5000) [inline src/main/java/com/faforever/client/patch/GameUpdaterImpl.java:85]: *did

File: src/main/java/com/faforever/client/game/GenerateMapController.java [hunk 2]
```diff
@@ -223,17 +223,17 @@ public void onGenerateMap() {
...
     } else if (cause instanceof OutdatedVersionException) {
-      notificationService.addImmediateErrorNotification(cause, "mapGenerator.tooOldVersion");
       log.warn("Map generation failed", e);
+      notificationService.addImmediateErrorNotification(cause, "mapGenerator.tooOldVersion");
     } else {
-      notificationService.addImmediateErrorNotification(e, "mapGenerator.generationFailed");
       log.warn("Map generation failed", e);
+      notificationService.addImmediateErrorNotification(e, "mapGenerator.generationFailed");
     }
   }
```
Comment (Brutus5000) [inline src/main/java/com/faforever/client/game/GenerateMapController.java]: All of these warning should either be moved outside the if block or customized to say more about the type of error.

**Review by Brutus5000 — CHANGES_REQUESTED** [review]
I like it

**Review by Brutus5000 — APPROVED** [review]

**Comment by Katharsas** [comment]
Nice

**Pull request merged**